### PR TITLE
do not refetch couch user

### PR DESCRIPTION
**Pull request by benrudolph**
@dannyroberts I'm mostly guessing that this is just legacy, but is there any reason the restore code re-fetches the couch user even though we have it set during middleware? found this being set a long time ago: https://github.com/dimagi/commcare-hq/commit/a6689bacfb749078562ed1f758f1cd0882e5639a

fyi: @dimagi/scale-team

### CODE REVIEW

File: corehq/apps/ota/views.py [hunk 1]
```diff
@@ -56,10 +56,8 @@ def restore(request, domain, app_id=None):
     We override restore because we have to supply our own
     user model (and have the domain in the url)
     """
-    couch_user = CouchUser.from_django_user_include_anonymous(domain, request.user)
-    assert couch_user is not None, 'No couch user to use for restore'
-    update_device_id(couch_user, request.GET.get('device_id'))
-    response, timing_context = get_restore_response(domain, couch_user, app_id, **get_restore_params(request))
+    update_device_id(request.couch_user, request.GET.get('device_id'))
+    response, timing_context = get_restore_response(domain, request.couch_user, app_id, **get_restore_params(request))
```
Comment (dimagimon) [inline corehq/apps/ota/views.py]: E501 line too long (118 > 115 characters)

**Review by snopoke — APPROVED** [review]

**Comment by snopoke** [comment]
@emord - good to get this in with NIC deploy (assuming build passes etc)

**Review by emord — APPROVED** [review]

File: corehq/apps/ota/views.py [hunk 2]
```diff
@@ -56,11 +56,9 @@ def restore(request, domain, app_id=None):
     We override restore because we have to supply our own
     user model (and have the domain in the url)
     """
-    couch_user = CouchUser.from_django_user_include_anonymous(domain, request.user)
-    assert couch_user is not None, 'No couch user to use for restore'
     if toggles.ENIKSHAY.enabled(domain):
-        update_device_id(couch_user, request.GET.get('device_id'))
-    response, timing_context = get_restore_response(domain, couch_user, app_id, **get_restore_params(request))
+        update_device_id(request.couch_user, request.GET.get('device_id'))
+    response, timing_context = get_restore_response(domain, request.couch_user, app_id, **get_restore_params(request))
```
Comment (dimagimon) [inline corehq/apps/ota/views.py]: E501 line too long (118 > 115 characters)

**Comment by emord** [comment]
@benrudolph I merged your other branch for enikshay and it conflicted. It's a simple conflict so i think it can be merged as long as the original test passes. Can you check my work?

**Comment by benrudolph** [comment]
yep looks good, thanks

**Comment by emord** [comment]
Thanks. I killed the build on the merge. The original build passed

**Pull request merged**